compiler/noirc_frontend/src/parser/parser/modifiers.rs

```diff
@@ -0,0 +1,38 @@
+use noirc_errors::Span;
+
+use crate::{ast::ItemVisibility, token::Keyword};
+
+use super::Parser;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub(crate) struct Modifiers {
+    pub(crate) visibility: ItemVisibility,
+    pub(crate) visibility_span: Span,
+    pub(crate) unconstrained: Option<Span>,
+    pub(crate) comptime: Option<Span>,
+    pub(crate) mutable: Option<Span>,
+}
+
+impl<'a> Parser<'a> {
+    /// Modifiers = 'unconstrained'? ItemVisibility 'comptime'? 'mut'?
```

Comment:
Sidenote: I still think `unconstrained` should be after `pub` but this would be a large breaking change.

Reply:
I agree with this change. I was quite surprised when I tried to make a `pub unconstrained` fn a while back and saw this.

We shouldn't do this in this PR but we could allow both but format into the pub-first format. We can restrict it again in a raft of 1.0 breaking changes.

Reply:
I agree too! I was going to ask about it last week in our channel, but I also didn't want to change things in this PR.

We could probably allow it `unconstrained` before and after `pub` to begin with, erroring if it's duplicate, then eventually only allow it after `pub`.